src/botPage/view/blockly/blocks/tools/webhook.js

```diff
@@ -0,0 +1,182 @@
+            connection.connect(itemBlock.previousConnection);
+            connection = itemBlock.nextConnection;
+        }
+        return containerBlock;
+    },
+    /**
+     * Reconfigure this block based on the mutator dialog's components.
+     * @param {!Blockly.Block} containerBlock Root block in mutator.
+     * @this Blockly.Block
+     */
+    compose(containerBlock) {
+        let itemBlock = containerBlock.getInputTargetBlock('STACK');
+        // Count number of inputs.
+        const connections = [];
+        while (itemBlock) {
+            connections.push(itemBlock.valueConnection_);
+            itemBlock = itemBlock.nextConnection && itemBlock.nextConnection.targetBlock();
+        }
+        // Disconnect any children that don't belong.
+        for (let i = 0; i < this.itemCount_; i++) {
+            const connection = this.getInput(`ADD${i}`).connection.targetConnection;
+            if (connection && connections.indexOf(connection) === -1) {
+                const targetBlock = connection.getSourceBlock();
+
+                connection.disconnect();
+                if (targetBlock.isShadow()) {
+                    targetBlock.dispose(false);
+                }
+            }
+        }
+        this.itemCount_ = connections.length;
+        this.updateShape_(true);
+        // Reconnect any child blocks.
+        for (let i = 0; i < this.itemCount_; i++) {
+            Blockly.Mutator.reconnect(connections[i], this, `ADD${i}`);
+        }
+    },
+    /**
+     * Store pointers to any connected child blocks.
+     * @param {!Blockly.Block} containerBlock Root block in mutator.
+     * @this Blockly.Block
+     */
+    saveConnections(containerBlock) {
+        let itemBlock = containerBlock.getInputTargetBlock('STACK');
+        let i = 0;
+        while (itemBlock) {
+            const input = this.getInput(`ADD${i}`);
+            itemBlock.valueConnection_ = input && input.connection.targetConnection;
+            i++;
+            itemBlock = itemBlock.nextConnection && itemBlock.nextConnection.targetBlock();
+        }
+    },
+    /**
+     * Modify this block to have the correct number of inputs.
+     * @private
+     * @this Blockly.Block
+     */
+    updateShape_(render) {
+        if (this.itemCount_ && this.getInput('EMPTY')) {
+            this.removeInput('EMPTY');
+        } else if (!this.itemCount_ && !this.getInput('EMPTY')) {
+            this.appendDummyInput('EMPTY').appendField(translate('Empty payload'));
+        }
+        let i;
+        for (i = 0; i < this.itemCount_; i++) {
+            if (!this.getInput(`ADD${i}`)) {
+                const input = this.appendValueInput(`ADD${i}`);
+
+                if (i === 0) {
+                    input.appendField(translate('Payload:'));
+                }
+
+                const { connection } = input;
+                const keypair = this.workspace.newBlock('key_value_pair', `keyvalue${i}`);
+                keypair.setShadow(true);
```

Comment:
This block doesn't have to be a shadow block, because the user won't be able to connect anything if it is.